### PR TITLE
Event Arena Controller

### PR DESCRIPTION
**Pull request by AnturK**
Event arena controller for you know what event.

TODO:

- [x] Turn lower/upper corner into just a/b and pick min/x/y
- [x] Unique HUD icon
- [ ] ~~Turn special actions into key list and turn interface to TGUI ? (Optional)~~ Another time.

### CODE REVIEW

File: code/game/machinery/computer/arena.dm [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+	/// Name of currently loaded template
+	var/current_arena_template = "None"
+	// What turf arena clears to
+	var/empty_turf_type = /turf/open/indestructible
+	// List of team ids
+	var/list/teams = list(ARENA_RED_TEAM,ARENA_GREEN_TEAM)
+	/// List of hud instances indedxed by team id
+	var/static/list/team_huds = list()
+	/// List of hud colors indexed by team id
+	var/static/list/team_colors = list(ARENA_RED_TEAM = "red", ARENA_GREEN_TEAM = "green")
```
Comment (Qustinnus) [inline code/game/machinery/computer/arena.dm]: I don't get the point of this list. They're both always the same value. Also this list isnt used.

File: code/game/machinery/computer/arena.dm [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+
+/obj/machinery/computer/arena/Initialize(mapload, obj/item/circuitboard/C)
+	. = ..()
+	LoadDefaultArenas()
+	GenerateAntagHuds()
+
+/obj/machinery/computer/arena/proc/GenerateAntagHuds()
+	for(var/team in teams)
+		var/datum/atom_hud/antag/teamhud = team_huds[team]
+		if(!teamhud) //These will be shared between arenas because this stuff is expensive and cross arena fighting is not a thing anyway
```
Comment (Qustinnus) [inline code/game/machinery/computer/arena.dm]: unnecesary nesting

File: code/game/machinery/computer/arena.dm [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+
+/// Handle arena specific special effects here.
+/obj/machinery/computer/arena/proc/arena_afterload(arena_filename)
+	/* Example code
+	switch(arena_filename)
+		if("example.dmm")
+			to_chat(world,"EXAMPLE ARENA FIGHT STARTING NOW. OR WHATEVER")
+		if("ripandtear.dmm")
+			give_everyone_chainsaws()
+	*/
```
Comment (Qustinnus) [inline code/game/machinery/computer/arena.dm]: Wouldn't it be better to use map template datums for this? Add procs there that get run? bit better OOP wise.

File: code/game/machinery/computer/arena.dm [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+/obj/machinery/computer/arena/proc/change_outfit(mob/user,team)
+	outfits[team] = user.client.robust_dress_shop()
+
+/obj/machinery/computer/arena/proc/toggle_spawn(mob/user)
+	ready_to_spawn = !ready_to_spawn
+	to_chat(user,"You [ready_to_spawn ? "enable" : "disable"] the spawners.")
+	log_admin("[key_name(user)] toggled event arena spawning for [arena_id] arena.")
+	// Could use update_icon on spawnpoints here to show they're on
+	if(ready_to_spawn)
```
Comment (Qustinnus) [inline code/game/machinery/computer/arena.dm]: gay nesting

File: code/game/machinery/computer/arena.dm [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+	for(var/team in teams)
+		var/obj/machinery/arena_spawn/team_spawn = get_spawn(team)
+		var/obj/effect/countdown/arena/A = new(team_spawn)
+		A.start()
+		countdowns += A
+
+/obj/machinery/computer/arena/proc/begin()
+	ready_to_spawn = FALSE
+	set_doors(closed = FALSE)
+	//I guess could add some sound here
```
Comment (Qustinnus) [inline code/game/machinery/computer/arena.dm]: do it then

File: code/game/machinery/computer/arena.dm (hunk 1 above)
Comment (AnturK) [reply code/game/machinery/computer/arena.dm]: Oh whoops i added table but forgot to use it.

**Review by ghost — CHANGES_REQUESTED** [review]
some small stuff

File: code/game/machinery/computer/arena.dm (hunk 5 above)
Comment (Qustinnus) [reply code/game/machinery/computer/arena.dm]: https://www.youtube.com/watch?v=-HOP7cQDmz4 here some samples u can use

**Pull request merged**